renderdoc/data/glsl/gl_texsample.h

```diff
@@ -51,17 +51,23 @@ vec3 CalcCubeCoord(vec2 uv, int face)
 
 // these bindings are defined based on the RESTYPE_ defines in debuguniforms.h
 
+#ifndef OPENGL_ES
 layout(binding = 1) uniform usampler1D texUInt1D;
-layout(binding = 2) uniform usampler2D texUInt2D;
-layout(binding = 3) uniform usampler3D texUInt3D;
+#endif
+layout(binding = 2) uniform PRECISION usampler2D texUInt2D;
+layout(binding = 3) uniform PRECISION usampler3D texUInt3D;
 // cube = 4
+#ifndef OPENGL_ES
 layout(binding = 5) uniform usampler1DArray texUInt1DArray;
-layout(binding = 6) uniform usampler2DArray texUInt2DArray;
+#endif
+layout(binding = 6) uniform PRECISION usampler2DArray texUInt2DArray;
 // cube array = 7
+#ifndef OPENGL_ES
 layout(binding = 8) uniform usampler2DRect texUInt2DRect;
-layout(binding = 9) uniform usamplerBuffer texUIntBuffer;
+#endif
+layout(binding = 9) uniform PRECISION usamplerBuffer texUIntBuffer;
 #ifdef GL_ARB_texture_multisample
-layout(binding = 10) uniform usampler2DMS texUInt2DMS;
+layout(binding = 10) uniform PRECISION usampler2DMS texUInt2DMS;
```

Comment:
This file is kind of 50:50 on whether there's more code shared or different between GLES and GL. I'm leaning towards splitting it into `gl_texsample.h` and `gles_texsample.h`, with e.g. `CalcCubeCoord` moved into `debuguniforms.h` or somewhere common. It would mean some duplicated code but also no #ifdefs so it would be much clearer which texture types are and aren't supported in each. What do you think?